### PR TITLE
feat: add favourites screen

### PR DESCRIPTION
**Pull request by abhishek-gkmit**
- add favourite screen context
- add `useFavourites` hook
- add storage functionality for favourites

### CODE REVIEW

File: src/config/contexts/FavouritesContext.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import {
+  getFavouriteRecipesFromStorage,
+  saveFavouriteRecipesToStorage,
+} from '@utility/asyncStorage';
+import React, { useState, createContext, useEffect, useCallback } from 'react';
```
Comment (adiTrivsme) [inline src/config/contexts/FavouritesContext.tsx]: fix import order

File: src/constants/index.ts [hunk 2]
```diff
@@ -26,4 +26,6 @@ const diets = [
 const ingredientImageUrl = (imageName: string) =>
   `https://img.spoonacular.com/ingredients_500x500/${imageName}`;
 
-export { colorSchemes, cuisines, diets, ingredientImageUrl };
+const FAVOURITES_RECIPES = 'favourites';
```
Comment (adiTrivsme) [inline src/constants/index.ts:29]: what is this?

Comment (abhishek-gkmit) [reply src/constants/index.ts:29]: it's storage key constant to use with `async storage` for saving favorite recipes in local storage

Comment (adiTrivsme) [reply src/constants/index.ts:29]: define all storage keys in single a object

**Review by adiTrivsme — APPROVED** [review]

**Pull request merged**